share/CMakeJS.cmake

```diff
@@ -0,0 +1,642 @@
+      OUTPUT_VARIABLE NODE_ADDON_API_DIR
+      # COMMAND_ERROR_IS_FATAL ANY
+    )
+    string(REGEX REPLACE "[\r\n\"]" "" NODE_ADDON_API_DIR "${NODE_ADDON_API_DIR}")
+    set(NODE_ADDON_API_DIR "${NODE_ADDON_API_DIR}" CACHE PATH "Node Addon API Headers directory." FORCE)
+
+    unset(NODE_ADDON_API_INC_FILES)
+    file(GLOB NODE_ADDON_API_INC_FILES "${NODE_ADDON_API_DIR}/*.h")
+    set(NODE_ADDON_API_INC_FILES "${NODE_ADDON_API_INC_FILES}" CACHE FILEPATH "Node Addon API Header files." FORCE)
+    source_group("Node Addon API (C++)" FILES "${NODE_ADDON_API_INC_FILES}")
+
+    if(VERBOSE)
+        message(STATUS "NODE_ADDON_API_DIR: ${NODE_ADDON_API_DIR}")
+    endif()
+endfunction()
+
+# Acquire if needed...
+if(NOT DEFINED NODE_ADDON_API_DIR)
+    cmakejs_acquire_napi_cpp_files()
+    message(STATUS "NODE_ADDON_API_DIR: ${NODE_ADDON_API_DIR}")
+    if(NOT DEFINED NODE_ADDON_API_INC_FILES)
+	file(GLOB NODE_ADDON_API_INC_FILES "${NODE_ADDON_API_DIR}/*.h")
+	set(NODE_ADDON_API_INC_FILES "${NODE_ADDON_API_INC_FILES}" CACHE FILEPATH "Node Addon API Header files." FORCE)
+        source_group("Node Addon API (C++)" FILES "${NODE_ADDON_API_INC_FILES}")
+    endif()
+endif()
+
+#[=============================================================================[
+Silently create an interface library (no output) with all Addon API dependencies
+resolved, for Addon targets to link with.
+
+(This should contain most of cmake-js globally-required configuration)
+
+Targets:
+
+cmake-js::node-dev
+cmake-js::node-api
+cmake-js::node-addon-api
+cmake-js::cmake-js
+
+]=============================================================================]#
+
+# NodeJS system installation headers
+# cmake-js::node-dev
+add_library                 (node-dev INTERFACE)
+add_library                 (cmake-js::node-dev ALIAS node-dev)
+target_include_directories  (node-dev INTERFACE "${CMAKE_JS_INC}")
+target_sources              (node-dev INTERFACE "${CMAKE_JS_SRC}")
+target_sources              (node-dev INTERFACE "${CMAKE_JS_INC_FILES}")
+target_link_libraries       (node-dev INTERFACE "${CMAKE_JS_LIB}")
+
+# Node API (C) - requires NodeJS system installation headers
+# cmake-js::node-api
+add_library                 (node-api INTERFACE)
+add_library                 (cmake-js::node-api ALIAS node-api)
+target_include_directories  (node-api INTERFACE "${NODE_API_HEADERS_DIR}")
+target_sources              (node-api INTERFACE "${NODE_API_INC_FILES}")
+target_link_libraries       (node-api INTERFACE cmake-js::node-dev)
+
+# Node Addon API (C++) - requires Node API (C)
+# cmake-js::node-addon-api
+add_library                 (node-addon-api INTERFACE)
+add_library                 (cmake-js::node-addon-api ALIAS node-addon-api)
+target_include_directories  (node-addon-api INTERFACE "${NODE_ADDON_API_DIR}")
+target_sources              (node-addon-api INTERFACE "${NODE_ADDON_API_INC_FILES}")
+target_link_libraries       (node-addon-api INTERFACE cmake-js::node-api)
```

Comment:
It looks like this library will exist even if `NODE_ADDON_API_DIR` failed to be found. Which could result in compliation failing due to missing headers instead of a more descriptive failure due to a missing library.

It is up to the user to install `node-addon-api` if they want it, it is entirely possible that they arent using it, so it missing should not be a hard failure unless they try to link/reference to it